### PR TITLE
Update Helm upgrade documentation

### PR DESCRIPTION
**Pull request by ykulazhenkov**
Add description how to apply manual rollingUpgrade
on the cluster if containerized OFED driver is used.

Follow up PR for #270 

Signed-off-by: Yury Kulazhenkov <ykulazhenkov@nvidia.com>

### CODE REVIEW

File: deployment/network-operator/README.md [hunk 1]
```diff
@@ -218,15 +218,45 @@ helm upgrade -n network-operator  network-operator mellanox/network-operator --v
 
 >__NOTE__: `--devel` option required if you want to use the beta release
 
-### Return PODs to the drained nodes
 
->__NOTE__: this operation required only if **containerized OFED** is in use and if drain command
-was applied before to remove PODs from the cluster nodes
+### Restart PODs with containerized OFED drive
```
Comment (moshe010) [inline deployment/network-operator/README.md:222]: s/drive/driver

File: deployment/network-operator/README.md [hunk 2]
```diff
@@ -218,15 +218,45 @@ helm upgrade -n network-operator  network-operator mellanox/network-operator --v
...
+
+#### Remove PODs with secondary network from the node
+When containerized OFED driver reloaded on the node, all PODs which use secondary network based on
+NVIDIA Mellanox NICs will lose network interface in their containers.
+To prevent outage you need to remove all PODs which use secondary network from the node
+before you reload the driver POD.
+
+This can be done with node drain command:
+```
+kubectl drain <NODE_NAME> --pod-selector=<selector_for_pods>
```
Comment (moshe010) [inline deployment/network-operator/README.md:239]: can you give example on how to use this command?
It expect to run this command per node?

File: deployment/network-operator/README.md [hunk 3]
```diff
@@ -218,15 +218,45 @@ helm upgrade -n network-operator  network-operator mellanox/network-operator --v
...
+
+#### Restart OFED driver POD
+Find OFED driver POD name for the node
+```
+kubectl get po -l driver-pod -o wide -A
+```
+
+Delete OFED driver POD from the node
+```
+kubectl delete po -n <DRIVER_NAMESPACE> <OFED_POD_NAME>
```
Comment (moshe010) [inline deployment/network-operator/README.md:250]: Why not 
1. drain all nodes (not matter what pods we have)\
2. restart all ofed containers 
I don't get what it give us if you need to run it per node?

File: deployment/network-operator/README.md (hunk 1 above)
Comment (ykulazhenkov) [reply deployment/network-operator/README.md]: done

File: deployment/network-operator/README.md (hunk 2 above)
Comment (ykulazhenkov) [reply deployment/network-operator/README.md:256]: Yes, this command should run per node. I will add details under another comment from you

File: deployment/network-operator/README.md (hunk 3 above)
Comment (ykulazhenkov) [reply deployment/network-operator/README.md:267]: It is required to implement rolling upgrade logic. In this case we don't need to remove all PODs with secondary networks from the cluster at the some time. I added some details to the doc itself.
We trying to achieve this flow:
1. Move PODs from node A to node B
2. Upgrade drive on node A
3. Return PODs to node A
4. Move PODs from node B to node A
5.  Upgrade drive on node B
So, we don't need to stop everething to upgrade the driver

Comment (adrianchiris) [reply deployment/network-operator/README.md:267]: nit : po -> pod

File: deployment/network-operator/README.md [hunk 4]
```diff
@@ -218,15 +218,62 @@ helm upgrade -n network-operator  network-operator mellanox/network-operator --v
...
-### Return PODs to the drained nodes
 
->__NOTE__: this operation required only if **containerized OFED** is in use and if drain command
-was applied before to remove PODs from the cluster nodes
+### Restart PODs with containerized OFED driver
+
+>__NOTE__: this operation required only if **containerized OFED** is in use
+
+Helm upgrade command will upgrade DaemonSet of the OFED driver to point to the new driver version,
+but will not remove or restart PODs with the driver on the nodes.
```
Comment (adrianchiris) [inline deployment/network-operator/README.md:227]: can you clarify why restart of pods wont happen ? e.g due to the daemonset update strategy, the daemonset will not remove or restart PODs with the driver on nodes.

atm it reads (to me) as if helm is doing it.

File: deployment/network-operator/README.md (hunk 3 above)
Comment (adrianchiris) [reply deployment/network-operator/README.md:267]: RE Moshe's comment, we could just provide both methods and let user decide.

Comment (ykulazhenkov) [reply deployment/network-operator/README.md]: I replaced all naming shortcust in kubectl commands with full resources names.
Also, I rephrased some sentences to more explicitly express that both options are possible.

File: deployment/network-operator/README.md (hunk 4 above)
Comment (ykulazhenkov) [reply deployment/network-operator/README.md]: I updated this section in the doc

File: deployment/network-operator/README.md [hunk 5]
```diff
@@ -218,15 +218,71 @@ helm upgrade -n network-operator  network-operator mellanox/network-operator --v
...
+```
+
+>__NOTE__: replace <NODE_NAME> with `-l "network.nvidia.com/operator.mofed.wait=false"` if you
+> want to drain all nodes at once
+
+
+#### Restart OFED driver POD
+Find OFED driver POD name for the node
+```
+kubectl get po -l app=mofed-<OS_NAME> -o wide -A
```
Comment (adrianchiris) [inline deployment/network-operator/README.md:267]: nit: po -> pod

File: deployment/network-operator/README.md [hunk 6]
```diff
@@ -218,15 +218,71 @@ helm upgrade -n network-operator  network-operator mellanox/network-operator --v
...
+```
+_example for Ubuntu 20.04: `kubectl get po -l app=mofed-ubuntu20.04 -o wide -A`_
```
Comment (adrianchiris) [inline deployment/network-operator/README.md:269]: nit: po -> pod

File: deployment/network-operator/README.md (hunk 5 above)
Comment (ykulazhenkov) [reply deployment/network-operator/README.md]: fixed

File: deployment/network-operator/README.md (hunk 6 above)
Comment (ykulazhenkov) [reply deployment/network-operator/README.md]: fixed

**Review by adrianchiris — APPROVED** [review]

**Review by moshe010 — APPROVED** [review]

**Pull request merged**